packages/playwright-core/src/server/recorder.ts

```diff
@@ -135,7 +135,8 @@ export class Recorder implements InstrumentationListener, IRecorder {
     this._contextRecorder.on(ContextRecorder.Events.Change, (data: { sources: Source[], primaryFileName: string }) => {
       this._recorderSources = data.sources;
       this._pushAllSources();
-      this._recorderApp?.setFileIfNeeded(data.primaryFileName);
+      if (this._mode === 'none')
+        this._recorderApp?.setFile(data.primaryFileName);
```

Comment:
This seems to always switch to the `primaryFileName`, even after user explicitly changed from "javascript" to "python" in the language picker. Am I missing something?

Reply:
When?